parameter_store_client_test.go

```diff
@@ -18,6 +19,9 @@ var param2 = new(ssm.Parameter).
 	SetValue("rds.something.aws.com").
 	SetARN("arn:aws:ssm:us-east-2:aws-account-id:/my-service/dev/DB_HOST")
 
+var putParameterInputReceived ssm.PutParameterInput
```

Comment:
please move this global variable inside the `stubSSMClient` struct like we do with the other methods and try to access it from there to assert

Reply:
I haven't done a _whole lot_ of Go, so bear with me here, please. When I put the variable inside of the `stubSSMClient`, I can write to it fine within `stubSSMClient.PutParameter()` (L65), but then when I try to access it anywhere afterwards, it is not available because the interface is more restrictive (`ssmClient` only has 3 methods defined for the interface) or I can cast the type, but then we lose the extra parts defined by the `stubSSMClient`, e.g. `stubSSMClient.PutParameterInputReceived`. 

I will updated the branch with what I have so far, though.